lesson11/main.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const body = document.getElementById("body");
+const wrapper = document.createElement("div");
+body.appendChild(wrapper);
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.src = "img/loading-circle.gif";
+   wrapper.appendChild(loadingIcon);
+};
+const endLoading = () => wrapper.remove();
+const jsonURL = "https://jsondata.okiba.me/v1/json/KKl45211103113313";
+
+async function fetchData() {
+   try {
+      const response = await fetch(jsonURL);
+      const json = await response.json();
+      return json.data;
+   } catch(error) {
+      alert('データの取得に失敗しました。')
+      console.error('エラー');
```

Comment:
```suggestion
   } catch(error) {
    throw new Error("データの取得に失敗しました。");
```
IMO
For example, if an error occurs in getting the data, the result of init will be undefined and the catch method will be executed, right?

In that case, I think the alert will appear twice, and the error will also appear twice in the console.
However, I think this is not a good idea if you think of it as a single coherent process.

Here we'll rethrow the error and
How about receiving it in init...?

Reply:
Thank you for your review.
I fixed the part you pointed out.

9874db9
I could not think enough about how the error should be handled.
Your comments were helpful and I'll do more research about this.

Reply:
It will be easier to understand if you try to make an error.
Please try it :)
```
// 適当にURLを変えてみる
const jsonURL = "https://jsondata.okiba.me/v1/json/";
```